### PR TITLE
Crosschainswaps V3

### PR DESCRIPTION
**Pull request by fringlesinthestreet**
As relay's dest address is an EOA, just forcing the `to` address as we used to do with socket will result in the tx going through (and possibly losing funds if it was _edited_), so now it throws an error instead.

### CODE REVIEW

File: sdk/package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "0.18.0",
+  "version": "0.19.0-beta.0",
```
Comment (fringlesinthestreet) [inline sdk/package.json:2]: Must change before merge

File: sdk/src/utils/constants.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { BigNumber } from '@ethersproject/bignumber';
 import { ChainId, EthereumAddress } from '../types';
 export const ETH_ADDRESS = '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE';
-export const API_BASE_URL = 'https://swap.p.rainbow.me';
+export const API_BASE_URL = 'https://swap.s.rainbow.me';
```
Comment (fringlesinthestreet) [inline sdk/src/utils/constants.ts:4]: Must change before merge

**Comment by fringlesinthestreet** [comment]
Needs:
- publish new swap SDK version on npm
- Update versions on APP & BX
- A changes on frontend logic to avoid generating an unnecessary approval tx

## APP

Need to replace [this line](https://github.com/rainbow-me/rainbow/blob/a64ad84893c9201250f73947f317e346c22d8a22/src/raps/unlockAndCrosschainSwap.ts#L28) with:  

```typescript
  const shouldNotHaveApproval = tradeDetails.no_approval !== undefined && tradeDetails.no_approval;
  if (!nativeAsset && routeAllowanceTargetAddress && !shouldNotHaveApproval) {
```

## BX 

Need to replace [this line](https://github.com/rainbow-me/browser-extension/blob/1f33946170e69e71344a18246da3ab6f9f95455e/src/core/raps/unlockAndCrosschainSwap.ts#L120) with:  

```typescript
  const shouldNotUnlockAsset = quote.no_approval !== undefined && quote.no_approval;
  if (!isNativeAssetUnwrapping && !nativeAsset && !shouldNotUnlockAsset) {
```

File: sdk/src/utils/sanity_check.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+ *   sellTokenAddress: '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE'
+ * };
+ * console.log(getToAddressFromCrosschainQuote(Source.CrosschainAggregatorRelay, quoteRelayETH));
+ * // Output: '0xabcdefabcdefabcdefabcdefabcdefabcdefabcdef'
+ *
+ * // Example for a quote from CrosschainAggregatorRelay where the sell token is not ETH
+ * const quoteRelayERC20 = {
+ *   to: '0xabcdefabcdefabcdefabcdefabcdefabcdefabcdef',
+ *   data: '0xa9059cbb000000000000000000000000f70da97812cb96acdf810712aa562db8dfa3dbef...',
+ *   sellTokenAddress: '0x1234567890123456789012345678901234567890'
```
Comment (welps) [inline sdk/src/utils/sanity_check.ts:85]: `to` should be the same as `sellTokenAddress` in this scenario, right?

**Review by welps — APPROVED** [review]

File: sdk/src/utils/sanity_check.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+ * @param data The hex encoded input data string from an ERC-20 transfer transaction. This string
+ * should include the method ID followed by the encoded parameters (address and amount).
+ *
+ * @returns { string | undefined } The destination address. If any error happens.
+ * Returns 'undefined' if it could not decode the call data.
+ */
+export function decodeERC20TransferToData(
+  data: string | undefined
+): string | undefined {
+  if (!data?.startsWith(ERC20_TRANSFER_SIGNATURE)) {
```
[the remark this answers is not in the document]
Comment (fringlesinthestreet) [reply sdk/src/utils/sanity_check.ts:125]: Yes

**Pull request merged**